### PR TITLE
Add global username

### PR DESCRIPTION
**Pull request by oSumAtrIX**
## About

This PR is an attempt to initiate adding global usernames according to the recently merged PR https://github.com/discord/discord-api-docs/pull/6218.

## TODO

~~Before merging this pull request, the added TODOs should be solved.~~ Additionally, the changes should be checked for completeness and tested roughly because I am unsure if I have changed everything necessary to support global usernames and it may be possible that a `get_global_username` function is missing or similar for example.

### CODE REVIEW

**Comment by mkrasnitski** [comment]
Actually, `DefaultAvatar` is no longer used anywhere in the codebase and should be removed. There was a discussion about it [here](https://discord.com/channels/381880193251409931/673965002805477386/1103018307495399424) but it was forgotten about.

File: src/model/user.rs [hunk 1]
```diff
@@ -187,7 +187,8 @@ impl CurrentUser {
 
 /// An enum that represents a default avatar.
 ///
-/// The default avatar is calculated via the result of `discriminator % 5`.
+/// The default avatar is calculated via the result of `(id >> 22) % 6`.
+/// Users on the legacy username system will continue using `discriminator % 5`.
```
Comment (oSumAtrIX) [inline src/model/user.rs]:
```suggestion
/// The default avatar is calculated via the result of `discriminator % 5`.
```

File: src/model/user.rs [hunk 2]
```diff
@@ -208,6 +209,9 @@ pub enum DefaultAvatar {
     /// The avatar when the result is `4`.
     #[serde(rename = "1cbd08c76f8af6dddce02c5138971129")]
     Red,
+    /// The avatar when the result is `5`.
+    #[serde(rename = "TODO")] // TODO: replace with actual hash
+    Pink,
```
Comment (oSumAtrIX) [inline src/model/user.rs]:
```suggestion
```

**Comment by GnomedDev** [comment]
`Member::display_name` should also be changed to prioritize the `global_name` over the `name`.

**Review by arqunis — APPROVED** [review]

**Comment by oSumAtrIX** [comment]
@arqunis I may note that it is quite highly that I have missed something.

**Comment by oSumAtrIX** [comment]
Was this meant to be closed?

**Comment by arqunis** [comment]
Shikes, sorry, wanted to add a commit onto your branch >.>. Will manually merge it.

**Comment by oSumAtrIX** [comment]
Ill reopen a follow up PR

**Comment by arqunis** [comment]
No need.

**Comment by arqunis** [comment]
Merged: https://github.com/serenity-rs/serenity/commit/1e4797aa8625d26dfda31a5d7b4305b004195060

**Pull request closed without merging**